gapis/resolve/memory.go

```diff
@@ -49,7 +49,10 @@ func filterTypedRanges(ranges []*service.TypedMemoryRange) []*service.TypedMemor
 	sort.Slice(ranges, func(i, j int) bool {
 		return (ranges[i].Root < ranges[j].Root ||
 			ranges[i].Root == ranges[j].Root &&
-				ranges[i].Type.TypeIndex < ranges[j].Type.TypeIndex)
+				ranges[i].Type.TypeIndex < ranges[j].Type.TypeIndex ||
+			ranges[i].Root == ranges[j].Root &&
+				ranges[i].Type.TypeIndex == ranges[j].Type.TypeIndex &&
+				ranges[i].Range.Base < ranges[j].Range.Base)
```

Comment:
This added sort criteria makes sure we see entries in order. Otherwise, later-entries that are encountered earlier may skip earlier-entries encountered later.